js/components/slider.js

```diff
@@ -35,21 +48,201 @@ var Slider = React.createClass({
+   * @return {Number} The next value to the left or right of the current value.
+   */
+  getNextSliderValue: function(forward) {
+    var value = 0;
+    var sign = forward ? 1 : -1;
+    var delta = Utils.ensureNumber(this.props.value) + (Utils.ensureNumber(this.props.step, 1) * sign);
+    var min = Utils.ensureNumber(this.props.minValue, -Infinity);
+    var max = Utils.ensureNumber(this.props.maxValue, Infinity);
+    value = Utils.constrainToRange(delta, min, max);
+    value = Utils.toFixedNumber(value, 2);
+    return value;
+  },
+
+  /**
+   * Uses the slider settings to generate the values for the different aria attributes
+   * associated with sliders. Will format values as a percentage if this.props.usePercentageForAria
+   * is set to true.
+   * @private
+   * @return {Object} An object with the following properties: valueMin, valueMax, valueNow, valueText.
+   */
+  getAriaValues: function() {
+    var aria = {};
+
+    if (this.props.usePercentageForAria) {
+      aria.valueMin = 0;
+      aria.valueMax = 100;
+      aria.valueNow = Utils.ensureNumber(this.props.value, 0) * 100 / Utils.ensureNumber(this.props.maxValue, 1);
+      aria.valueText = CONSTANTS.ARIA_LABELS.SLIDER_VALUE_TEXT;
+      aria.valueText = aria.valueText.replace(MACROS.PERCENT, aria.valueNow).replace(MACROS.SETTING, this.props.settingName);
+    } else {
+      aria.valueMin = this.props.minValue;
+      aria.valueMax = this.props.maxValue;
+      aria.valueNow = this.props.value;
+      aria.valueText = aria.valueNow + ' ' + this.props.settingName;
+    }
+    return aria;
   },
 
   render: function() {
+    var aria = this.getAriaValues();
+
     return (
-      <input type="range" className={this.props.className} min={this.props.minValue} max={this.props.maxValue}
-           value={this.props.value} step={this.props.step} ref={this.props.itemRef} role={this.props.role}
-           onChange={this.changeValue} onClick={this.changeValue} onMouseMove={this.changeValue}/>
+      <input
+        type="range"
+        ref={this.props.itemRef}
+        className={ClassNames('oo-slider', this.props.className)}
+        min={this.props.minValue}
+        max={this.props.maxValue}
+        value={this.props.value}
+        step={this.props.step}
+        data-focus-id={this.props.focusId}
+        tabIndex="0"
+        aria-label={this.props.ariaLabel}
+        aria-valuemin={aria.valueMin}
+        aria-valuemax={aria.valueMax}
+        aria-valuenow={aria.valueNow}
+        aria-valuetext={aria.valueText}
+        role={CONSTANTS.ARIA_ROLES.SLIDER}
+        onMouseDown={this.onMouseDown}
+        onMouseUp={this.onMouseUp}
+        onChange={this.changeValue}
+        onClick={this.changeValue}
+        onMouseMove={this.onMouseMove}
+        onKeyDown={this.isIeFixRequired() ? this.onKeyDown : null}/>
     );
   }
 });
+
+Slider.defaultProps = {
+  focusId: Math.random().toString(36).substr(2, 10),
+  usePercentageForAria: false,
+};
+
```

Comment:
Do we need to include propTypes for Slider?

Reply:
Adding..